akvo/rsr/filters.py

```diff
@@ -98,26 +98,45 @@ def get_locations(location, locations):
 def location_choices(qs):
     """From a queryset get possible location filter choices"""
 
-    if qs.model is Project:
-        location_model = ProjectLocation
-    elif qs.model is ProjectUpdate:
+    country_ids = (
+        get_recipient_country_ids(qs) if qs.model is Project else
```

Comment:
I think we should concatenate the country IDs from recipient country and project_location. Otherwise we exclude around 100 projects from being found when filtering on country.

```sql
select p.id from rsr_project p, rsr_publishingstatus s where p.id not in (select DISTINCT p.id from rsr_project p, rsr_recipientcountry r where p.id = r.project_id) and p.id = s.project_id and s.status = 'published'
```